File: .backend/Dopa.Api/Models/ResourceArticle.cs
using System;

namespace Dopa.Api.Models;

public class ResourceArticle : BaseEntity
{
    public string Title { get; set; } = string.Empty;
    public string Summary { get; set; } = string.Empty;
    public string Content { get; set; } = string.Empty;
    public string Category { get; set; } = string.Empty;
    public string SourceUrl { get; set; } = string.Empty;
    public DateTime PublishedAt { get; set; } = DateTime.UtcNow;
}

File: .app/lib/models/resource_article.dart
class ResourceArticle {
  ResourceArticle({
    required this.id,
    required this.title,
    required this.summary,
    required this.category,
    required this.sourceUrl,
  });

  final String id;
  final String title;
  final String summary;
  final String category;
  final String sourceUrl;

  factory ResourceArticle.fromJson(Map<String, dynamic> json) => ResourceArticle(
        id: json['id'] as String,
        title: json['title'] as String,
        summary: json['summary'] as String,
        category: json['category'] as String,
        sourceUrl: json['sourceUrl'] as String,
      );
}
